### PR TITLE
feat(services): implement Timeline service module

### LINKED ISSUE #28: Build Timeline service module (CRUD + collaborator + junction management)
## Description
Create the Timeline service module implementing all PostgREST-style CRUD operations via the Supabase client. Timelines are the top-level organizational entity.

## Functions (`lib/services/timeline-service.ts`)
- `getTimelines(filters?)` — List with optional visibility/owner/search filters, pagination
- `getTimelineBySlug(slug)` — Single fetch with related data (events, periods, collaborators)
- `getTimelineById(id)` — Single fetch by UUID
- `createTimeline(data)` — Insert with Zod validation, slug generation, owner assignment
- `updateTimeline(id, data)` — Partial update with Zod validation
- `deleteTimeline(id)` — Soft delete or hard delete (CASCADE removes events)
- `publishTimeline(id)` / `unpublishTimeline(id)` — Toggle visibility between draft/public
- `addCollaborator(timelineId, userId, role)` — Insert into timeline_collaborators
- `removeCollaborator(timelineId, userId)` — Delete from timeline_collaborators
- `updateCollaboratorRole(timelineId, userId, role)` — Update role (editor/viewer)
- `getCollaborators(timelineId)` — List collaborators with profile data
- `addEventToTimeline(timelineId, eventId, sortOrder)` — Junction: timeline_events
- `removeEventFromTimeline(timelineId, eventId)` — Junction delete
- `addMediaToTimeline(timelineId, mediaId)` — Junction: timeline_media

## Patterns
- Use `supabase.from('timelines').select().eq()` PostgREST pattern
- RLS handles authorization — no manual auth checks in service code
- Return typed results using generated Supabase types
- Handle Supabase errors uniformly (wrap in Result type or throw)
- Validate inputs with Zod schemas before database calls

## Acceptance Criteria
- [ ] All CRUD operations work against Supabase
- [ ] Zod validation runs before every write operation
- [ ] Slug auto-generated from title on create
- [ ] Collaborator management (add/remove/update role) works
- [ ] Junction table management for events and media
- [ ] Error handling returns meaningful messages
- [ ] TypeScript types are fully strict (no `any`)

## Dependencies
- Phase 0 schema (#13, #14 — tables exist)
- #21 (Supabase client configured)
- #23 (Zod schemas)
- #25 (Slug utility)

## References
- system-design.md §5.2 Client Service Layer
- system-design.md §4.3 timelines table DDL
- PRD §3.1 Timeline management requirements

### PR DESCRIPTION
**Pull request by shaes-farm**
## Summary

Implements the Timeline service module as specified in #28.

## Changes

- **`packages/services/src/modules/timeline-service.ts`** — 14 exported functions covering full CRUD, publish/unpublish, collaborator management, and junction table management for timelines
- **`packages/services/src/modules/timeline-service.test.ts`** — 41 tests (100% stmt/line/func, 92.3% branch coverage) using chainable `vi.fn()` builder mocks
- **`packages/services/vitest.config.ts`** — `timeline-service.ts` added to the coverage `include` list to enforce the 80% threshold

## Design Decisions

- **Dependency injection**: each function accepts a `SupabaseClient<Database>` as its first argument — callers (RSC, Route Handlers, Server Actions) supply their own client; mocking in tests is trivial with no module-level side effects
- **Throw on error**: errors surface via `assertNoError()` helper, consistent with the Zod-throws pattern already used by `TemporalService`
- **Slug auto-generation**: `createTimeline` derives the slug from `title` via `generateSlug` + `resolveCollision`, scoped to the authenticated user's existing slugs — no caller-supplied slug required

## ⚠️ Decision Needed

`addEventToTimeline` omits the `sortOrder` parameter referenced in the issue spec — the `timeline_events` junction table has no `sort_order` column (unlike `timeline_media`). A `// DECISION NEEDED` comment is left in the source. Either the spec should be updated or a migration adding the column should be filed.

## Acceptance Criteria

- [x] All CRUD operations work against Supabase
- [x] Zod validation runs before every write operation
- [x] Slug auto-generated from title on create
- [x] Collaborator management (add/remove/update role) works
- [x] Junction table management for events and media
- [x] Error handling returns meaningful messages
- [x] TypeScript types are fully strict (no `any`)

## Validation

All checks pass (`pnpm verify`): format ✓ · lint ✓ · types ✓ · 242 tests pass · build ✓

Closes #28

### CODE REVIEW

File: packages/services/src/modules/timeline-service.ts [hunk 1]
```diff
@@ -0,0 +1,418 @@
+import type { SupabaseClient } from "@supabase/supabase-js";
+import { z } from "zod";
+import { timelineSchema } from "../schemas/timeline.js";
+import type { TimelineInput } from "../schemas/timeline.js";
+import { generateSlug, resolveCollision } from "../utils/slug.js";
+import type { Database } from "../supabase/types.js";
```
[the remark this answers is not in the document]
Comment (shaes-farm) [reply packages/services/src/modules/timeline-service.ts:6]: Fixed in commit 3feb823 — added packages/services/src/timeline-service.ts which re-exports everything from ./modules/timeline-service.js, making the module importable as @repo/services/timeline-service per the package.json exports map.

File: packages/services/src/modules/timeline-service.ts [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+ */
+export async function getTimelineById(
+  client: SupabaseClient<Database>,
+  id: string,
+): Promise<TimelineWithRelations> {
+  const { data, error } = await client
+    .from("timelines")
+    .select(
+      "*, timeline_collaborators(*), timeline_events(*), timeline_media(*)",
+    )
```
[the remark this answers is not in the document]
Comment (shaes-farm) [reply packages/services/src/modules/timeline-service.ts:124]: Acknowledged — the current junction tables (timeline_events, timeline_media) only store foreign key IDs; there are no separate event or media detail tables in the schema yet. Once those tables exist, the select can be expanded to embed the full related entities. Leaving this as a known limitation until issue #28 follow-on work adds those tables.

File: packages/services/src/modules/timeline-service.ts [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+ * Returns a page of timelines, optionally filtered by visibility, owner, or
+ * full-text search against the `search_vector` generated column.
+ */
+export async function getTimelines(
+  client: SupabaseClient<Database>,
+  filters: TimelineFilters = {},
+): Promise<TimelineRow[]> {
+  const { visibility, userId, search, page = 1, pageSize = 20 } = filters;
+  const from = (page - 1) * pageSize;
+  const to = from + pageSize - 1;
```
[the remark this answers is not in the document]
Comment (shaes-farm) [reply packages/services/src/modules/timeline-service.ts]: Fixed in commit 3feb823 — page is now clamped to Math.max(1, ...) and pageSize to Math.min(100, Math.max(1, ...)). Three new tests cover page=0, pageSize=0, and pageSize=999 edge cases.

File: packages/services/src/modules/timeline-service.ts [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+
+  // Full validation via Zod (includes slug now that we've generated it)
+  const validated = timelineSchema.parse({ ...data, slug });
+
+  type TimelineInsert = Database["public"]["Tables"]["timelines"]["Insert"];
+  const { data: row, error } = await client
+    .from("timelines")
+    .insert({ ...(validated as unknown as TimelineInsert), user_id: user.id })
+    .select()
+    .single();
```
[the remark this answers is not in the document]
Comment (shaes-farm) [reply packages/services/src/modules/timeline-service.ts]: Fixed in commit 3feb823 — createTimeline now retries up to 3 times on PostgreSQL unique-violation error code 23505, appending a random 4-char hex suffix to the slug on each retry. This guards against the TOCTOU race between slug availability check and insert.

File: packages/services/src/modules/timeline-service.ts [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+ */
+export async function getCollaborators(
+  client: SupabaseClient<Database>,
+  timelineId: string,
+): Promise<CollaboratorRow[]> {
+  const { data, error } = await client
+    .from("timeline_collaborators")
+    .select("*")
+    .eq("timeline_id", timelineId);
+
```
[the remark this answers is not in the document]
Comment (shaes-farm) [reply packages/services/src/modules/timeline-service.ts:324]: Addressed in commit 3feb823 — updated JSDoc to explicitly state that profile data is not available because timeline_collaborators has no FK to a public profiles table in the current schema, callers must fetch profiles separately by user_id, and this is a known limitation pending a profiles table or auth.users view.

File: packages/services/src/modules/timeline-service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (shaes-farm) [reply packages/services/src/modules/timeline-service.ts:6]: Fixed in commit 3feb823 — added `packages/services/src/timeline-service.ts` which re-exports everything from `./modules/timeline-service.js`, making the module importable as `@repo/services/timeline-service` per the package.json exports map.
